resolvers/node/resolution.rs

```diff
@@ -1838,6 +1853,25 @@ impl<
     )
   }
 
+  /// Attempts to read the "module" field from a package.json for cases where
+  /// a package only defines a `module` entrypoint without a `main`.
+  /// Note: this is a non-standard Node resolution behavior used to support
+  /// npm packages that only expose ESM builds via a `module` field.
+  fn get_module_field(&self, package_json: &PackageJson) -> Option<String> {
+    // read the package.json file and attempt to parse a `module` field
+    let file_bytes = match self.sys.fs_read(&package_json.path) {
+      Ok(bytes_cow) => bytes_cow,
+      Err(_) => return None,
+    };
```

Comment:
> Because deno_package_json::PackageJson doesn't expose module, legacy_main_resolve now reads and parses the package.json to find module in this case.

As noted, this works, but is not ideal because it hits the file system again. A better fix would be to update deno_package_json and add this field.